xc/common/utils/prjxray_routing_import.py

```diff
@@ -733,6 +734,68 @@ def import_graph_nodes(conn, graph, node_mapping, connection_box_map):
             assert False, side
 
 
+def remove_graph_nodes_outside_roi(conn, node_mapping, roi):
```

Comment:
We shouldn't remove graph nodes outside the ROI.  There is already (potentially broken) logic for not adding edges to nodes used in the ROI.

Reply:
https://github.com/SymbiFlow/symbiflow-arch-defs/blob/6e6e8c7ce9e6f2f553ebf6610714e9b3b9e7d20c/xc/common/utils/prjxray_edge_library.py#L1120-L1125

Reply:
My suspicion is that bidirectional edges are bypassing this logic, because the src can become the destination when creating the reverse edge.

Reply:
> We shouldn't remove graph nodes outside the ROI

Indeed, the naming of this function is actually wrong. The intent with this was to remove graph_nodes within the ROI that can potentially have connections with nodes outside of the ROI.